### PR TITLE
Add Churrosoft Deck-8 and RGB variant.

### PR DESCRIPTION
**Pull request by Polsaker**
## Description

Add Churrosoft's Deck-8 macropad and its RGB variant.

Note: The hardware availability page is not live yet but is going up in a few days.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/churrosoft/deck8/rgb/info.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "max_brightness": 200,
+    "layout": [
+      {"flags": 4, "matrix": [0, 0], "x": 0, "y": 0},
+      {"flags": 4, "matrix": [0, 1], "x": 1, "y": 0},
+      {"flags": 4, "matrix": [0, 2], "x": 2, "y": 0},
+      {"flags": 4, "matrix": [0, 3], "x": 3, "y": 0},
+      {"flags": 4, "matrix": [1, 0], "x": 0, "y": 1},
+      {"flags": 4, "matrix": [1, 1], "x": 1, "y": 1},
+      {"flags": 4, "matrix": [1, 2], "x": 2, "y": 1},
+      {"flags": 4, "matrix": [1, 3], "x": 3, "y": 1}
```
Comment (filterpaper) [inline keyboards/churrosoft/deck8/rgb/info.json:24]:
```suggestion
      {"flags": 4, "matrix": [0, 0], "x":  45, "y": 21},
      {"flags": 4, "matrix": [0, 1], "x":  90, "y": 21},
      {"flags": 4, "matrix": [0, 2], "x": 135, "y": 21},
      {"flags": 4, "matrix": [0, 3], "x": 180, "y": 21},
      {"flags": 4, "matrix": [1, 0], "x":  45, "y": 42},
      {"flags": 4, "matrix": [1, 1], "x":  90, "y": 42},
      {"flags": 4, "matrix": [1, 2], "x": 135, "y": 42},
      {"flags": 4, "matrix": [1, 3], "x": 180, "y": 42}
```

The physical x and y positions of LEDs are limited to a range of [224 by 64](https://docs.qmk.fm/#/feature_rgb_matrix?id=common-configuration). You can consider scaling the LEDs in this manner to provide sufficient breadth for RGB animation.

**Review by filterpaper — CHANGES_REQUESTED** [review]

File: keyboards/churrosoft/deck8/rgb/info.json (hunk 1 above)
Comment (Polsaker) [reply keyboards/churrosoft/deck8/rgb/info.json]: Sweet. Just tested these changes with the spash animation and it's way better

File: keyboards/churrosoft/deck8/info.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  "maintainer": "Polsaker",
+  "usb": {
+    "vid": "0x1BE2",
+    "pid": "0xC100",
+    "device_version": "1.0.0"
+  },
+  "bootmagic": {
+    "enabled": true,
+    "matrix": [0, 0]
+  },
```
Comment (zvecr) [inline keyboards/churrosoft/deck8/info.json:14]:
```suggestion
```

File: keyboards/churrosoft/deck8/info.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  "build": {
+    "lto": true
+  },
+  "features": {
+    "command": false,
```
Comment (zvecr) [inline keyboards/churrosoft/deck8/info.json:19]:
```suggestion
    "bootmagic": true,
    "command": false,
```

File: keyboards/churrosoft/deck8/info.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{
+  "keyboard_name": "Deck-8",
+  "manufacturer": "Churrosoft",
+  "url": "https://churrosoft.ar/deck",
+  "maintainer": "Polsaker",
+  "usb": {
+    "vid": "0x1BE2",
```
Comment (zvecr) [inline keyboards/churrosoft/deck8/info.json:7]: You generally shouldnt use an existing vendor ID that is already allocated.

**Review by zvecr — CHANGES_REQUESTED** [review]
The folder structure is somewhat incorrect for 2 variants, you need to have
```
keyboards/churrosoft/deck8/<something>/rules.mk
keyboards/churrosoft/deck8/rgb/rules.mk
```

and remove `keyboards/churrosoft/deck8/rules.mk`.

File: keyboards/churrosoft/deck8/info.json (hunk 4 above)
Comment (Polsaker) [reply keyboards/churrosoft/deck8/info.json:7]: Good catch, I'm changing it to 0xCBBC which doesn't seem to be allocated according to [this](https://www.usb.org/sites/default/files/vendor_ids082119_0.pdf)

**Comment by Polsaker** [comment]
Applied the changes. I also moved the `readme.md` to the new `noleds` directory, since it seems to me it makes more sense if it's there

File: keyboards/churrosoft/deck8/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        _______, _______, _______, _______,
+        _______, _______, _______, _______
+        ),
+    [2] = LAYOUT(
+        _______, _______, _______, _______,
+        _______, _______, _______, _______
+        ),
+    [3] = LAYOUT(
+        _______, _______, _______, _______,
+        _______, _______, _______, _______
```
Comment (waffle87) [inline keyboards/churrosoft/deck8/keymaps/via/keymap.c:34]:
```suggestion
```

File: keyboards/churrosoft/deck8/rgb/config.h [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+#pragma once
+
+#define RGB_MATRIX_LED_COUNT 8
+#define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_SOLID_COLOR
+#define RGB_MATRIX_DEFAULT_HUE 152
+#define RGB_MATRIX_DEFAULT_SAT 232
+#define RGB_MATRIX_DEFAULT_VAL 180
+#define RGB_MATRIX_LED_FLUSH_LIMIT 16
```
Comment (waffle87) [inline keyboards/churrosoft/deck8/rgb/config.h:24]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by zvecr — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**